cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -146,7 +146,24 @@ func (mcm *mcmCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.N
 		return nil, nil
 	}
 
-	return mcm.mcmManager.GetMachineDeploymentForMachine(ref)
+	md, err := mcm.mcmManager.GetMachineDeploymentForMachine(ref)
+	if err != nil {
+		return nil, err
+	}
+
+	var isManaged bool
+	for _, nodeGroup := range mcm.machinedeployments {
+		if nodeGroup.Id() == md.Id() {
+			isManaged = true
+			break
+		}
+	}
```

Comment:
```suggestion
	for _, managedMachineDeployment := range mcm.machinedeployments {
		if managedMachineDeployment.Id() == md.Id() {
			isManaged = true
			break
		}
	}
```
Does this make sense to keep it more aligned with the object type? WDYT?